client/ayon_core/addon/interfaces.py

```diff
@@ -117,77 +147,85 @@ def get_inventory_action_paths(self, host_name):
 
         Args:
            host_name (str): For which host are the plugins meant.
-        """
 
+        """
         return self._get_plugin_paths_by_type("inventory")
 
 
 class ITrayAddon(AYONInterface):
     """Addon has special procedures when used in Tray tool.
 
-    IMPORTANT:
-    The addon. still must be usable if is not used in tray even if
-    would do nothing.
-    """
+    Important:
+        The addon. still must be usable if is not used in tray even if it
+        would do nothing.
 
+    """
     tray_initialized = False
-    _tray_manager = None
+    manager: AddonsManager = None
+    _tray_manager: TrayManager = None
     _admin_submenu = None
 
     @abstractmethod
-    def tray_init(self):
+    def tray_init(self) -> None:
         """Initialization part of tray implementation.
 
         Triggered between `initialization` and `connect_with_addons`.
 
         This is where GUIs should be loaded or tray specific parts should be
-        prepared.
-        """
+        prepared
 
-        pass
+        """
+        raise NotImplementedError
 
     @abstractmethod
-    def tray_menu(self, tray_menu):
+    def tray_menu(self, tray_menu: QtWidgets.QMenu) -> None:
```

Comment:
```suggestion
    def tray_menu(self, tray_menu: "QtWidgets.QMenu") -> None:
```

Reply:
you don't need this because of the `__future__.annotations`